File: src/0344-Reverse-String/reverse_string.cpp
#include <iostream>
#include <vector>

class Solution {
public:
    void reverseString(std::vector<char>& s) {
        int left = 0;
        int right = s.size() - 1;

        while (left < right) {
            std::swap(s[left], s[right]);
            left++;
            right--;
        }
    }
};

int main() {
    std::vector<char> s = {'h','e','l','l','o'};

    Solution solution;
    solution.reverseString(s);

    for (int i = 0; i < s.size(); i++){
        std::cout << s[i] << " ";
    }
    
    return 0;
}

File: src/0344-Reverse-String/reverse_string.py
from typing import List

class Solution:
    def reverseString(self, s: List[str]) -> None:
        """
        Do not return anything, modify s in-place instead.
        """
        left: int = 0
        right: int = len(s) - 1

        while left < right:
            s[left] , s[right] = s[right], s[left]
            left += 1
            right -= 1

if __name__ == "__main__":
    s: List[str] = ["h","e","l","l","o"]
    
    solution = Solution()
    solution.reverseString(s)
    print(s)
    